application/controllers/ServicesController.php

```diff
@@ -198,6 +201,91 @@ public function searchEditorAction()
         $this->setTitle(t('Adjust Filter'));
     }
 
+    public function gridAction()
```

Comment:
This action is still missing some processing for the searchbar. Try adding a filter with it, you should notice it isn't applied. The searchbar requires specific handling by an action, check what the `servicesAction` does with it and perform the same here.